### PR TITLE
Adding skipped reason for test-cases

### LINKED ISSUE #68: Add detailed reason when skipping/xfailing/xpassing
It would be great if the `reason` args could be filled when tests are skipped/xfailed/xpassing.
Something like "Skipped due to issue XXX-1111 being unresolved"

### PR DESCRIPTION
**Pull request by lukas-bednar**
Partially fixes #68

### CODE REVIEW

**Comment by lukas-bednar** [comment]
This PR is WorkInProgress, so far it doesn't behave consistently for all cases xfail/xpass/skipped.
It needs tests covering all cases.

File: pytest_jira.py [hunk 1]
```diff
@@ -79,14 +87,24 @@ def pytest_runtest_makereport(self, item, call):
...
                         rep.outcome = "skipped"
                     elif PYTEST_MAJOR_VERSION < 3:
                         rep.outcome = "failed"
-                    rep.wasxfail = "failed"
+                    rep.wasxfail = self._get_skipped_reson(issue_id)
+#                    item.add_report_section(
+#                        rep.when, "stdout", self._get_skipped_reson(issue_id)
+#                    )
                     break
+                else:
```
Comment (liiight) [inline pytest_jira.py]: You got an indentation issue there

**Comment by lukas-bednar** [comment]
I was able to get into into report for **skipped** and **xfailed** status, but I can not achieve it for **xpassed**.

```
<testsuite errors="0" failures="0" name="pytest" skips="2" tests="3" time="0.053">
  <testcase classname=".tox.py27.tmp.testdir.test_jira_report_issue_details0.test_jira_report_issue_details" file=".tox/py27/tmp/testdir/test_jira_report_issue_details0/test_jira_report_issue_details.py" line="2" name="test_skipped" time="0.000269889831543">
    <skipped message="JIRA[ORG-1382] Issue https://issues.jboss.org/browse/ORG-1382" type="pytest.skip">/home/lbednar/work/pytest_jira/pytest_jira.py:127: &lt;py._xmlgen.raw object at 0x7f6ac0675610&gt;</skipped>
  </testcase>
  <testcase classname=".tox.py27.tmp.testdir.test_jira_report_issue_details0.test_jira_report_issue_details" file=".tox/py27/tmp/testdir/test_jira_report_issue_details0/test_jira_report_issue_details.py" line="6" name="test_xpassed" time="0.000361680984497"/>
  <testcase classname=".tox.py27.tmp.testdir.test_jira_report_issue_details0.test_jira_report_issue_details" file=".tox/py27/tmp/testdir/test_jira_report_issue_details0/test_jira_report_issue_details.py" line="10" name="test_xfail" time="0.000775098800659">
    <skipped message="expected test failure">JIRA[ORG-1382] Issue
https://issues.jboss.org/browse/ORG-1382</skipped>
  </testcase>
</testsuite>
```

so it doesn't cover all requirements in #68 issue ...

**Comment by lukas-bednar** [comment]
Hi @liiight ,

I did some progress here, but still it will not show issue for **xpassed** tests.
Feel free to take a look ...

Lukas.

**Comment by liiight** [comment]
I'll try to take a peek.

**Comment by lukas-bednar** [comment]
Closing in favor of #91

**Pull request closed without merging**